### PR TITLE
feat: check_plain function support atom key

### PR DESCRIPTION
**Pull request by Rory-Z**

### CODE REVIEW

**Review by z8674558 — DISMISSED** [review]

File: src/hocon_schema.erl [hunk 1]
```diff
@@ -365,6 +365,8 @@ str(S) when is_list(S) -> S.
 bin(A) when is_atom(A) -> atom_to_binary(A, utf8);
 bin(S) -> iolist_to_binary(S).
 
+atom(B) when is_binary(B) -> binary_to_atom(B, utf8).
```
Comment (zmstone) [inline src/hocon_schema.erl:368]: to existing atom ?

**Review by zmstone — CHANGES_REQUESTED** [review]

File: src/hocon_schema.erl (hunk 1 above)
Comment (Rory-Z) [reply src/hocon_schema.erl]: Fixed

File: src/hocon_schema.erl [hunk 2]
```diff
@@ -767,7 +770,10 @@ do_plain_get([], Conf) ->
     %% value as-is
     Conf;
 do_plain_get([H | T], Conf) ->
-    Child = maps:get(H, Conf, undefined),
+    Child = case maps:get(H, Conf, undefined) of
```
Comment (zmstone) [inline src/hocon_schema.erl:773]: Catch error if it’s not a an existing atom, return undefined

**Comment by zmstone** [comment]
Please ensure 100% test coverage

File: src/hocon_schema.erl [hunk 3]
```diff
@@ -767,7 +770,10 @@ do_plain_get([], Conf) ->
     %% value as-is
     Conf;
 do_plain_get([H | T], Conf) ->
-    Child = maps:get(H, Conf, undefined),
```
Comment (zmstone) [inline src/hocon_schema.erl:770]: Not this pr’s issue, but: we need another clause to handle when Conf is not a map.

**Pull request closed without merging**